camel/toolkits/function_tool.py

```diff
@@ -247,11 +256,17 @@ def __init__(
         use_schema_assistant: Optional[bool] = False,
         schema_assistant_model: Optional[BaseModelBackend] = None,
         schema_generation_max_retries: int = 2,
+        synthesis_mode: Optional[bool] = False,
+        synthesis_assistant_model: Optional[BaseModelBackend] = None,
+        response_format: Optional[Type[BaseModel]] = None,
     ) -> None:
         self.func = func
         self.openai_tool_schema = openai_tool_schema or get_openai_tool_schema(
             func
         )
+        self.synthesis_mode = synthesis_mode
+        self.synthesis_assistant_model = synthesis_assistant_model
+        self.response_format = response_format
```

Comment:
Should get the `synthesize_output_format` from the signature of `self.func` if `synthesize_output_format=None`.